internal/dynamicserver/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/alecthomas/kong"
+	"google.golang.org/grpc"
+	pref "google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+var version = "v0.0.0"
+
+type config struct {
+	Address string           `help:"hostname:port" default:"localhost:9090"`
+	Version kong.VersionFlag `short:"V" help:"Print version information" group:"Other:"`
+
+	Service string `help:"Service name" default:"echo.EchoService"`
+	Method  string `help:"Method name" default:"Hello"`
+	In      string `help:"String field of request" default:"message"`
+	Out     string `help:"String field of response proto" default:"response"`
+
+	inMessagDescriptor  pref.MessageDescriptor
+	outMessagDescriptor pref.MessageDescriptor
+}
+
+func main() {
+	cfg := newConfig()
+	ctx := kong.Parse(cfg, kong.Vars{"version": version})
+	fmt.Println("starting server server on", cfg.Address)
+	err := run(cfg)
+	ctx.FatalIfErrorf(err)
+}
+
+func newConfig() *config {
+	in := HelloRequest{}
+	out := HelloResponse{}
+	return &config{
+		inMessagDescriptor:  in.ProtoReflect().Descriptor(),
```

Comment:
I think the in/out temp vars actually make this harder to read.
```
inMessagDescriptor:  HelloRequest{}.ProtoReflect().Descriptor(),
outMessagDescriptor: HelloResponse{}.ProtoReflect().Descriptor(),
```

I can see immediately what that is. With the vars, I have to scan up in the code to interpret it.

The in/out prefix on the field name already conveys the direction, but also the standard naming of MethodRequest/MethodResponse does that too.

Reply:
ok, agreed. fixed.

Reply:
oops. i actually mean to say: I tried this but it doesn't work:

```
./main.go:41:39: cannot call pointer method on HelloRequest{}
./main.go:42:40: cannot call pointer method on HelloResponse{}
```

Reply:
Stupid Go not allowing pointers to literals. Can't do `x := &3` either. Stupid Go.

If we were using go tip, we could do:
```
func ptr[T any](v T) *T {
  return &v
}
```

and then say
```
inMessagDescriptor:  ptr(HelloRequest{}).ProtoReflect().Descriptor(),
```
but that doesn't look that good anyway.

Reply:
At least this will go away when we use .pb files. I initially assumed that's why you had the vars - you would replace the `HelloRequest{}` in the var with code that looks up the registry.